test/e2e/setup/setup.go

```diff
@@ -164,10 +169,33 @@ func deployOperator() error {
 	); err != nil {
 		return errors.Errorf("error building operator deployment: %s", err)
 	}
+
+	if err := wait.PollImmediate(time.Second, time.Duration(30)*time.Second, hasDeploymentRequiredReplicas(dep)); err != nil {
+		return errors.New("error building operator deployment: the deployment does not have the required replicas")
+	}
+
 	fmt.Println("Successfully created the operator Deployment")
 	return nil
 }
 
+// hasDeploymentRequiredReplicas returns a condition function that indicates whether the given deployment
+// currently has the required amount of replicas in the ready state as specified in spec.replicas
+func hasDeploymentRequiredReplicas(dep *appsv1.Deployment) wait.ConditionFunc {
+	return func() (bool, error) {
+		err := e2eutil.TestClient.Get(context.TODO(),
+			types.NamespacedName{Name: dep.Name,
+				Namespace: e2eutil.OperatorNamespace},
+			dep)
+		if err != nil {
+			return false, errors.Errorf("error getting operator deployment: %s", err)
+		}
+		if dep.Status.ReadyReplicas == *dep.Spec.Replicas {
+			return true, nil
+		}
+		return false, nil
```

Comment:
nit: what do you think about replacing this with something like this
```
depNamespacedName := types.NamespacedName{Name: dep.Name, Namespace: e2eutil.OperatorNamespace}
if err := e2eutil.TestClient.Get(context.TODO(), depNamespacedName, dep); err != nil {
    return false, errors.Errorf("error getting operator deployment: %s", err)
}
replicasReachedTheGoalState := dep.Status.ReadyReplicas == *dep.Spec.Replicas
return replicasReachedTheGoalState, nil
```
it's really minor but it improves a bit the readability

Reply:
> replicasReachedTheGoalState := dep.Status.ReadyReplicas == *dep.Spec.Replicas

I think it's better to be explicit here about the if check, also to align with the existing pattern in the codebase -- I am inclined towards having multiple explicit returns instead of assigning it to a temporary variable.

Reply:
I suggested a temp variable to express the intention, but agree consistency is more important